lib/puppet/network/http/response.rb

```diff
@@ -5,7 +5,19 @@ def initialize(handler, response)
   end
 
   def respond_with(code, type, body)
-    @handler.set_content_type(@response, type)
+    format = Puppet::Network::FormatHandler.format_for(type)
+    mime = format.mime
+    charset = format.charset
+
+    if charset
+      if body.is_a?(String) && body.encoding != charset
+        body.encode!(charset)
+      end
+
+      mime += "; charset=#{charset.name.downcase}"
+    end
+
+    @handler.set_content_type(@response, mime)
```

Comment:
The good thing is that `format_to_mime` passes through the argument if the argument isn't a Format object. So this PR won't break puppetserver, as it will continue calling `format_to_mime("application/json; charset=utf-8")` and our `format_to_mime` will be a noop.